### PR TITLE
Add return to raise_for_status.

### PR DESCRIPTION
**Pull request by heckad**
Add return to raise_for_status for writing `resp = requests.get('SOME_URL').raise_for_status()` and same code.

### CODE REVIEW

**Comment by sargun** [comment]
Any reason why this didn't make progress? I quite like this.

**Comment by heckad** [comment]
I don`t know why tests failing. I am usied wsl for testing and all good. Now I will try again.

**Comment by heckad** [comment]
How to run tests corectly on windows 10?

**Comment by JARVIS-AI** [comment]
Do some test command with your changes 
also
take some auto test package and see what its result 
no matter what is your OS
read a python guide to know what your doing.

**Comment by heckad** [comment]
All tests comleat suddenly.

**Comment by heckad** [comment]
@kennethreitz, will be merged?

**Comment by JARVIS-AI** [comment]
> 
> 
> @kennethreitz, will be merged?

Review your work 
something is not right

**Comment by heckad** [comment]
> something is not right

Please tell me what is not right?

**Comment by heckad** [comment]
How long does it take?

**Comment by Andru1999** [comment]
@sargun please add thumbs up for this pull request. @kenneth-reitz for some reason ignores it.

**Comment by sargun** [comment]
Can you rebase and squash?

File: requests/models.py [hunk 1]
```diff
@@ -938,6 +938,8 @@ def raise_for_status(self):
 
         if http_error_msg:
             raise HTTPError(http_error_msg, response=self)
+
```
Comment (sargun) [inline requests/models.py:942]: Please add a comment to the docstring, something like "alternatively, returns self on success"

**Comment by frostming** [comment]
IMO `raise_for_status` is more like a no-return function. Chaining evaluation is good but it returning the response surprised me.

**Comment by heckad** [comment]
As you propose to call the function?

File: requests/models.py (hunk 1 above)
Comment (heckad) [reply requests/models.py:942]: Add the comment in docstring

**Comment by kennethreitz** [comment]
I agree with @frostming.

**Pull request closed without merging**